discovery/src/main/java/com/ibm/watson/developer_cloud/discovery/v1/Discovery.java

```diff
@@ -10,445 +10,705 @@
+  private static final String URL = "https://gateway.watsonplatform.net/discovery/api";
+
+  private String versionDate;
+
+  /** The Constant VERSION_DATE_2016_12_01. */
+  public static final String VERSION_DATE_2016_12_01 = "2016-12-01";
+
+  /**
+   * Instantiates a new `Discovery`.
+   *
+   * @param versionDate The version date (yyyy-MM-dd) of the REST API to use. Specifying this value will keep your API
+   *          calls from failing when the service introduces breaking changes.
+   */
+  public Discovery(String versionDate) {
+    super(SERVICE_NAME);
+    if ((getEndPoint() == null) || getEndPoint().isEmpty()) {
+      setEndPoint(URL);
+    }
+
+    Validator.isTrue((versionDate != null) && !versionDate.isEmpty(),
+        "'version cannot be null. Use " + VERSION_DATE_2016_12_01);
+
+    this.versionDate = versionDate;
+  }
+
+  /**
+   * Instantiates a new `Discovery` with username and password.
+   *
+   * @param versionDate The version date (yyyy-MM-dd) of the REST API to use. Specifying this value will keep your API
+   *          calls from failing when the service introduces breaking changes.
+   * @param username the username
+   * @param password the password
+   */
+  public Discovery(String versionDate, String username, String password) {
+    this(versionDate);
+    setUsernameAndPassword(username, password);
+  }
+
+  /**
+   * Create a collection.
+   *
+   * @param createCollectionOptions the {@link CreateCollectionOptions} containing the options for the call
+   * @return the {@link Collection} with the response
+   */
+  public ServiceCall<Collection> createCollection(CreateCollectionOptions createCollectionOptions) {
+    Validator.notNull(createCollectionOptions, "createCollectionOptions cannot be null");
+    RequestBuilder builder = RequestBuilder.post(String.format("/v1/environments/%s/collections",
+        createCollectionOptions.environmentId()));
+    builder.query(VERSION, versionDate);
+    final JsonObject contentJson = new JsonObject();
+    contentJson.addProperty("name", createCollectionOptions.name());
+    if (createCollectionOptions.description() != null) {
+      contentJson.addProperty("description", createCollectionOptions.description());
+    }
+    if (createCollectionOptions.language() != null) {
+      contentJson.addProperty("language", createCollectionOptions.language());
+    }
+    if (createCollectionOptions.configurationId() != null) {
+      contentJson.addProperty("configuration_id", createCollectionOptions.configurationId());
+    }
+    builder.bodyJson(contentJson);
+    return createServiceCall(builder.build(), ResponseConverterUtils.getObject(Collection.class));
+  }
+
+  /**
+   * Delete a collection.
+   *
+   * @param deleteCollectionOptions the {@link DeleteCollectionOptions} containing the options for the call
+   * @return the service call
+   */
+  public ServiceCall<Void> deleteCollection(DeleteCollectionOptions deleteCollectionOptions) {
```

Comment:
We decided to adopt the pattern of returning void for all deleteX methods, since any error on the delete should generate an exception.  @germanattanasio was in favor of this change.  But please state your case if you think this was an unwise decision.